unified_planning/grpc/proto_writer.py

```diff
@@ -0,0 +1,341 @@
+        return "exists"
+    elif op == "FORALL":
+        return "forall"
+
+    raise ValueError(f"Unknown operator `{op}`")
+
+
+class ProtobufWriter(Converter):
+    @handles(unified_planning.model.Fluent)
+    def _convert_fluent(self, fluent):
+        name = fluent.name()
+        sig = [
+            unified_planning_pb2.Parameter(name="", type=str(t))
+            for t in fluent.signature()
+        ]
+        valType = str(fluent.type())
+        return unified_planning_pb2.Fluent(
+            name=name, value_type=valType, parameters=sig
+        )
+
+    @handles(unified_planning.model.Object)
+    def _convert_object(self, obj):
+        return unified_planning_pb2.ObjectDeclaration(
+            name=obj.name(), type=obj.type().name()
+        )
+
+    @handles(unified_planning.model.FNode)
+    def _convert_fnode(self, exp):
+        payload = exp._content.payload
+        args = exp._content.args
+        atom = None
+        p_type = ""
+        kind = unified_planning_pb2.ExpressionKind.Value("UNKNOWN")
+        arg_list = []
+
+        # TODO: add variable support
+
+        if isinstance(payload, bool):
+            atom = unified_planning_pb2.Atom(boolean=payload)
+            kind = unified_planning_pb2.ExpressionKind.Value("CONSTANT")
+            p_type = "bool"
+        elif isinstance(payload, int):
+            atom = unified_planning_pb2.Atom(int=payload)
+            kind = unified_planning_pb2.ExpressionKind.Value("CONSTANT")
+            p_type = "int"
+        elif isinstance(payload, float):
+            atom = unified_planning_pb2.Atom(float=payload)
+            kind = unified_planning_pb2.ExpressionKind.Value("CONSTANT")
+            p_type = "real"
+        elif isinstance(payload, str):
+            atom = unified_planning_pb2.Atom(symbol=payload)
+        elif isinstance(payload, unified_planning.model.Fluent):
+            kind = unified_planning_pb2.ExpressionKind.Value("FLUENT_SYMBOL")
+            atom = unified_planning_pb2.Atom(symbol=payload.name())
+            p_type = str(payload.type())
+        elif isinstance(payload, unified_planning.model.Object):
+            # TODO: check if first element is fluent symbol
+            kind = unified_planning_pb2.ExpressionKind.Value("STATE_VARIABLE")
+            atom = unified_planning_pb2.Atom(symbol=payload.name())
+        elif isinstance(payload, unified_planning.model.ActionParameter):
+            kind = unified_planning_pb2.ExpressionKind.Value("PARAMETER")
+            p_type = str(payload.type())
+            atom = unified_planning_pb2.Atom(symbol=payload.name())
+        else:
+            kind = unified_planning_pb2.ExpressionKind.Value("FUNCTION_SYMBOL")
+            atom = unified_planning_pb2.Atom(symbol=map_operator(exp.node_type()))
+
+        for arg in args:
+            arg_list.append(self.convert(arg))
+
+        return unified_planning_pb2.Expression(
+            atom=atom, list=arg_list, kind=kind, type=p_type
```

Comment:
Here it seems that the `atom` and `arg_list` are not exclusive (same as comment on `arg_list` name).